accepted/0022-git-buildpack.md

```diff
@@ -0,0 +1,61 @@
+# Git Support
+
+## Proposal
+
+Introduce an (optional) Git buildpack into each language buildpack family. The buildpack
+will be charged with providing `git` metadata as build-time and run-time environment
+variables.
+
+## Motivation
+
+Applications of any language may have a need for information extracted from
+`git` metadata. For example, the commit sha might be used as versioning
+information when building a Go binary or used when running a Rails application.
+
+## Implementation
+
+A new `git` buildpack will be developed to detect whether the `.git` directory, and then
+read the directory to extract the following environment variables so that they
```

Comment:
Could we also add an env var for the git branch and git tag (if present)? My company injects those two items (plus the sha) into go binaries.

Reply:
One more: knowing if a git repo is "dirty" with `git status --porcelain` would also be a helpful ENV var (for conditionally applying something based on whether or not there are uncommitted changes).

Reply:
It's seeming likely that the set of env vars set by this buildpack would grow and grow with subsequent feature requests for user's specific needs. Is there a way to capture all of the information in the `.git` directory directly?

Reply:
The contents of the concourse git resource might provide good prior art on exactly which data might be useful.

https://github.com/concourse/git-resource#additional-files-populated